protocols/frontend/cloudstate/entity_key.proto

```diff
@@ -28,3 +28,18 @@ option go_package = "github.com/cloudstateio/go-support/cloudstate/;cloudstate";
 extend google.protobuf.FieldOptions {
     bool entity_key = 50002;
 }
+extend google.protobuf.MessageOptions {
+    EntityMessageOptions entity = 50002;
+}
+
+extend google.protobuf.OneofOptions {
+    bool api_response_value = 50002;
+}
+
+message EntityMessageOptions {
```

Comment:
Aren't these all these values mutually exclusive? I.e. a `oneof` or `enum` would be more fitting?

Reply:
That's correct, thanks for noticing. I threw these bools in here a bit haphazardly, not knowing whether or not the idea of introducing new / additional metadata through extensions is something we're open to... Happy to iterate on / refine / further explore this if there's interest in this direction 👍

Reply:
It could spin-off into it's own discussion, but I wouldn't want to have movement on the Scala language support be entirely hinged on that if that discussion is projected to take a long time. One idea to consider is that we take a first pass at implementing Scala support with the cloudstate API defined exactly as it currently is, while at the same time trying to move forward the discussion around new metadata. In coming up with my ideal example the ideas were hard to separate, and I do find them to be very closely related, so my preference would be to have a first-pass that includes the introduction of some new metadata, as long as that doesn't have the potential to draw out into a super long process.

Reply:
@ryanhanks I think you're right—let's keep the "new possible metadata" as a separate track—makes the Scala support a bit more focused.